core/core.go

```diff
@@ -304,6 +373,10 @@ func NewTextile(ctx context.Context, conf Config) (*Textile, error) {
 		if conf.Hub {
 			hpb.RegisterAPIServiceServer(t.server, hs)
 			upb.RegisterAPIServiceServer(t.server, us)
+			healthRpc.RegisterRPCServiceServer(t.server, &healthRpc.UnimplementedRPCServiceServer{})
+			netRpc.RegisterRPCServiceServer(t.server, &netRpc.UnimplementedRPCServiceServer{})
+			ffsRpc.RegisterRPCServiceServer(t.server, &ffsRpc.UnimplementedRPCServiceServer{})
+			walletRpc.RegisterRPCServiceServer(t.server, &walletRpc.UnimplementedRPCServiceServer{})
```

Comment:
For Hub gRPC server to process requests to powergate, we have to register _any_ service implementations, even if they aren't used. So I use the generated "unimplemented" services.

Reply:
Right, if you don't register these, you gRPC will return an error to the client before your interceptors are even called.